habitat_sim/agent/controls/pyrobot_noisy_controls.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import attr
+import magnum as mn
+import numpy as np
+
+import habitat_sim.bindings as hsim
+from habitat_sim import utils
+
+from .controls import ActuationSpec, SceneNodeControl, register_move_fn
+
+__all__ = ["PyRobotNoisyActuationSpec"]
+
+
+@attr.s(auto_attribs=True)
+class _MultivariateGaussian:
+    mean: np.array
+    cov: np.array
+
+    def __attrs_post_init__(self):
+        self.mean = np.array(self.mean)
+        self.cov = np.array(self.cov)
+        if len(self.cov.shape) == 1:
+            self.cov = np.diag(self.cov)
+
+
+@attr.s(auto_attribs=True)
+class MotionNoiseModel:
+    linear: _MultivariateGaussian
+    rotation: _MultivariateGaussian
+
+
+@attr.s(auto_attribs=True)
+class ControllerNoiseModel:
+    linear_motion: MotionNoiseModel
+    rotational_motion: MotionNoiseModel
+
+
+@attr.s(auto_attribs=True)
+class RobotNoiseModel:
+    ILQR: ControllerNoiseModel
+    Proportional: ControllerNoiseModel
+    Movebase: ControllerNoiseModel
+
+    def __getitem__(self, key):
+        return getattr(self, key)
+
+
+r"""
+Parameters contributed from PyRobot
+https://pyrobot.org/
+https://github.com/facebookresearch/pyrobot
+
+Please cite PyRobot if you use this noise model
+"""
+pyrobot_noise_models = {
+    "LoCoBot": RobotNoiseModel(
+        ILQR=ControllerNoiseModel(
+            linear_motion=MotionNoiseModel(
+                _MultivariateGaussian([0.014, 0.009], [0.006, 0.005]),
```

Comment:
Abstracting out the constants into a different class would make this code more readable.

Reply:
Maybe stick this in a different file?  Any class that has all these constants will be as unreadable as this class by definition (all these classes are doing is holding these constants).